kuka_sunrise_fri_driver/config/ros2_controller_config.yaml

```diff
@@ -12,6 +12,8 @@ controller_manager:
       type: kuka_controllers/FRIStateBroadcaster
     event_broadcaster:
       type: kuka_controllers/EventBroadcaster
+    external_torque_broadcaster:

```

Comment:
does this need a separate controller with the same type as the joint state broadcaster? would that not work to simply extend that config?

Reply:
Not really, because the joint state broadcaster only broadcasts joint position velocity and effort. You can remap different hardware interface to the joint state (what i have done here with the external torque) but you cannot extend the state structure itself. 
Maybe I could create a completely new broadcaster very similar to the `jointstate` one with new message types and everything, but then it will not be compatible with other ROS nodes that might use the standard `jointstate` message.

Reply:
okay, thanks, you should definitely not create a new joint_state broadcaster, that would ruin compatibility, so your solution seems to be the right way
just to make sure that I understand corectly: you publish the joint state as previously on the joint_state topic and additionally publish a JointState message on the state topic, where the torque values are the external torques but all other values are the same?

Reply:
Almost. The only data in it published to the "external_torque_broadcaster/joint_states" topic is a joint state message with the joint names and their external torques. The config yaml needs two extra lines to also broadcast the positions.

Reply:
great, sounds good :)